app/controllers/api/v2/registration_controller.rb

```diff
@@ -31,42 +32,46 @@ def global
-        param "#{facet_config.name}_attributes".to_sym, Hash, desc: facet_config.api_param_group_description || (N_("Parameters for host's %s facet") % facet_config.name) do
-          facet_config.host_configuration.load_api_controller
-          param_group facet_config.host_configuration.api_param_group, facet_config.host_configuration.api_controller
+      param :host, Hash, required: true, action_aware: true do
+        param :name, String, required: true
+        param :location_id, :number, required: true
+        param :organization_id, :number, required: true
+        param :ip, String, desc: N_("IPv4 address, not required if using a subnet with DHCP proxy")
+        param :ip6, String, desc: N_("IPv6 address, not required if using a subnet with DHCP proxy")
+        param :mac, String, desc: N_("required for managed host that is bare metal, not required if it's a virtual machine")
+        param :domain_id, :number, desc: N_("required if host is managed and value is not inherited from host group")
+        param :operatingsystem_id, :number, desc: N_("required if host is managed and value is not inherited from host group")
+        param :subnet_id, :number, desc: N_("required if host is managed and value is not inherited from host group")
+        param :model_id, :number
+        param :hostgroup_id, :number
+        param :host_parameters_attributes, Array, desc: N_("Host's parameters (array or indexed hash)") do
+          param :name, String, desc: N_("Name of the parameter"), required: true
+          param :value, String, desc: N_("Parameter value"), required: true
+          param :parameter_type, Parameter::KEY_TYPES, desc: N_("Type of value")
+          param :hidden_value, :bool
+        end
+        param :build, :bool
+        param :enabled, :bool, desc: N_("Include this host within Foreman reporting")
+        param :managed, :bool, desc: N_("True/False flag whether a host is managed or unmanaged. Note: this value also determines whether several parameters are required or not")
+        param :comment, String, desc: N_("Additional information about this host")
+        param :interfaces_attributes, Array, desc: N_("Host's network interfaces.") do
+          param_group :interface_attributes, ::Api::V2::InterfacesController
+        end
+        Facets.registered_facets.values.each do |facet_config|
+          next unless facet_config.host_configuration.api_param_group && facet_config.host_configuration.api_controller
+          param "#{facet_config.name}_attributes".to_sym, Hash, desc: facet_config.api_param_group_description || (N_("Parameters for host's %s facet") % facet_config.name) do
+            facet_config.host_configuration.load_api_controller
+            param_group facet_config.host_configuration.api_param_group, facet_config.host_configuration.api_controller
+          end
         end
       end
+      param :setup_insights, :bool, desc: N_("Set 'host_registration_insights' parameter for the host.")
```

Comment:
I'd add a bit more info the the API description. E.g.

_Set 'host_registration_insights' parameter for the host. If it is set to true, insights client will be installed and registered on Red Hat family operating systems._